mail_tracking/models/mail_message.py

```diff
@@ -95,6 +105,18 @@ def _get_email_cc(self):
             })
         return res
 
+    @api.multi
+    def _get_failed_message(self):
+        res = {}
+        for message in self:
+            res.update({
+                message.id: message.track_needs_action
+                and any(message.mail_tracking_ids.filtered(
+                    lambda x: x.state in ['error', 'rejected', 'spam',
```

Comment:
Appart from what's commented above, this will make it faster:

```suggestion
                    lambda x: x.state in {'error', 'rejected', 'spam',
```